src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyCacheCapacityAction.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions;
+
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.HEAP;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.ResourceEnum;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ModifyCacheCapacityAction implements Action {
+  public static final String NAME = "modifyCacheCapacity";
+  public static final int COOL_OFF_PERIOD_IN_SECONDS = 300;
+
+  private long currentCapacityInBytes;
+  private long desiredCapacityInBytes;
+  private ResourceEnum cacheType;
+  private NodeKey esNode;
+
+  private Map<ResourceEnum, Integer> upperBound = new HashMap<>();
+
+  public ModifyCacheCapacityAction(
+          NodeKey esNode, ResourceEnum cacheType, long currentCapacityInBytes, boolean increase) {
+    // TODO: Also consume NodeConfigurationRca
+    setBounds();
+    // TODO: Update the step size to be a percentage of max size instead of an absolute value
+    int STEP_SIZE_IN_BYTES = 50;
```

Comment:
should we make this step size a function of cache capacity ? 50 bytes is pretty small compared with the actual cache size

Reply:
Yes right. It will be a percentage of cache capacity. Presently, I have added a TODO. Will address this in the subsequent PR.